mitosheet/mitosheet/tests/user/test_user.py

```diff
@@ -277,4 +278,25 @@ def test_upgrades_version_3_to_version_4():
 def test_main_cli_turn_off_logging():
     initialize_user()
     subprocess.run(['python', '-m', 'mitosheet', 'turnofflogging'])
-    assert get_user_field(UJ_MITOSHEET_TELEMETRY) == False
\ No newline at end of file
+    assert get_user_field(UJ_MITOSHEET_TELEMETRY) == False
+
+def test_upgrades_to_new_experiment():
+    write_fake_user_json({
+        UJ_STATIC_USER_ID: 'github_action',
+        UJ_MITOSHEET_TELEMETRY: True,
+        UJ_MITOSHEET_PRO: False,
+        UJ_EXPERIMENT: {
+            'experiment_id': 'an_old_experiment',
+            'variant': 'A',
+        }
+    })
+
+    initialize_user()
+    check_user_json(user_email='', mitosheet_experiment_id=get_new_experiment()['experiment_id'])
```

Comment:
Hmm okay this is my understanding:

- Say I am a user who is onboarded on Jul 1, I will be on user.json version 6. 
- When I upgraded Mito, I will get upgraded to user.json version 7
- In that upgrade process, my user.json will now include a full onboarding_checklist (to ensure that it doesn't pop up), and I will also randomly get added to the experiment in line `206` of `upgrade.py`. 
- Since I've been added to the experiment, all of my future events will include the log parameters `experiment_id`, `experiment_installer_communication_and_time_to_value
`, `experiment_variant`.

This isn't necessarily going to throw off our Mixpanel reports. But it will make them fragile because we will have to be sure to always filter out the users who were not part of the A/B test. 

Alternativly, if we are upgrading a user from user.json 6 to 7 we could either not give them an experiment_id (this would require that the only time we add someone to an experiment is through the install or upgrade process, which seems like it should be the case), or assign them to the experiment_variant called "not included"

What do you think? It seems worth fixing up so we don't make our analyses more fragile in a relm of new analytics work already.

Reply:
I think there are ways we want to iterate this infrastructure, but it's hard to say exactly how it's best to do this. I think for now, I want to avoid introducing more complexity into this infrastructure and rather just filter by when the user was first seen or finished signup by date. 

I added fields for just tracking when the experiments start and end to the experiment tracker. I think that this is a good enough solution for now, especially given how few A/B tests we're running and the few analytics we're looking at!